service/history/archivalQueueTaskExecutor.go

```diff
@@ -0,0 +1,208 @@
+	return &archivalQueueTaskExecutor{
+		archiver:       archiver,
+		shardContext:   shardContext,
+		workflowCache:  workflowCache,
+		logger:         logger,
+		metricsHandler: metricsHandler,
+	}
+}
+
+func (e *archivalQueueTaskExecutor) Execute(
+	ctx context.Context,
+	executable queues.Executable,
+) (tags []metrics.Tag,
+	isActive bool, err error) {
+	task := executable.GetTask()
+	taskType := queues.GetArchivalTaskTypeTagValue(task)
+	tags = []metrics.Tag{
+		getNamespaceTagByID(e.shardContext.GetNamespaceRegistry(), task.GetNamespaceID()),
+		metrics.TaskTypeTag(taskType),
+	}
+	switch task := task.(type) {
+	case *tasks.ArchiveExecutionTask:
+		err = e.processArchiveExecutionTask(ctx, task)
+	default:
+		err = fmt.Errorf("task with invalid type sent to archival queue: %+v", task)
+	}
+	return tags, true, err
+}
+
+func (e *archivalQueueTaskExecutor) processArchiveExecutionTask(ctx context.Context, task *tasks.ArchiveExecutionTask) (err error) {
+	weContext, release, err := getWorkflowExecutionContextForTask(ctx, e.workflowCache, task)
+	if err != nil {
+		return err
+	}
+	defer func() { release(err) }()
+
+	mutableState, err := weContext.LoadMutableState(ctx)
+	if err != nil {
+		return err
+	}
+	if mutableState == nil || mutableState.IsWorkflowExecutionRunning() {
+		return nil
+	}
+	lastWriteVersion, err := mutableState.GetLastWriteVersion()
+	if err != nil {
+		return err
+	}
+	namespaceEntry := mutableState.GetNamespaceEntry()
+	err = CheckTaskVersion(
+		e.shardContext,
+		e.logger,
+		namespaceEntry,
+		lastWriteVersion,
+		task.Version,
+		task,
+	)
+	if err != nil {
+		return err
+	}
+
+	namespaceName := namespaceEntry.Name()
+	nextEventID := mutableState.GetNextEventID()
+	executionInfo := mutableState.GetExecutionInfo()
+	workflowTypeName := executionInfo.GetWorkflowTypeName()
+
+	executionState := mutableState.GetExecutionState()
+	memo := getWorkflowMemo(copyMemo(executionInfo.Memo))
+	closeTime, err := mutableState.GetWorkflowCloseTime(ctx)
+	if err != nil {
+		return err
+	}
+	if closeTime == nil {
+		return errors.New("can't archive workflow with nil close time")
+	}
+	branchToken, err := mutableState.GetCurrentBranchToken()
+	if err != nil {
+		return err
+	}
+
+	var targets []archival.Target
+	if e.shardContext.GetArchivalMetadata().GetVisibilityConfig().ClusterConfiguredForArchival() &&
+		namespaceEntry.VisibilityArchivalState().State == enumspb.ARCHIVAL_STATE_ENABLED {
+		targets = append(targets, archival.TargetVisibility)
+	}
+	if e.shardContext.GetArchivalMetadata().GetHistoryConfig().ClusterConfiguredForArchival() &&
+		namespaceEntry.HistoryArchivalState().State == enumspb.ARCHIVAL_STATE_ENABLED {
+		targets = append(targets, archival.TargetHistory)
+	}
+
+	historyURI, err := carchiver.NewURI(namespaceEntry.HistoryArchivalState().URI)
+	if err != nil {
+		e.logger.Error(
+			"Failed to parse history URI.",
+			tag.ArchivalURI(namespaceEntry.HistoryArchivalState().URI),
+			tag.Error(err),
+		)
+		return nil
```

Comment:
I know right now we just give up, but I think the point of durable archival and having a separate archival queue is in those situations, we don't need to worry about blocking other queues and can do infinite retry.
So I'd rather let the task keep retrying and trigger some alerts, then after the configuration is fixed, those tasks can complete, and we won't skip archival for any customer workflow.

Later when we have DLQ, we can put those tasks into DLQ.